src/quill-cursors/cursor.ts

```diff
@@ -106,6 +132,28 @@ export default class Cursor {
     selections.forEach((selection: ClientRect) => this._addSelection(selection, container));
   }
 
+  private _setHoverState(): void {
+    this._caretEl.classList.add(Cursor.CONTAINER_NO_POINTER_CLASS);
+    document.addEventListener('mouseover', this._toggleOpenedCursor);
+  }
+
+  private _toggleOpenedCursor(e: MouseEvent): void {
+    const shouldShow = this.toggleNearCursor(e.clientX, e.clientY);
+    this._caretEl.classList.toggle(Cursor.CONTAINER_NO_POINTER_CLASS, shouldShow);
+    document.removeEventListener('mouseover', this._toggleOpenedCursor);
+    if (!shouldShow) {
+      this._showFlagTimeout = null;
+      return;
+    }
+
+    this._showFlagTimeout = setTimeout(this._setHoverState, 100);
+  }
+
+  private _updateCoordinates(): void {
+    const {x, y, width, height} = this._caretEl.getBoundingClientRect();
```

Comment:
I think `getBoundingClientRect()` is an expensive function — rather than proactively update and track our coordinates, I think we should just ask for them when needed, so:

1. Remove the `coordinates` property
2. Change `_updateCoordinates(): void` -> `_getCoordinates(): ICoordinates`
3. Replace references to `this.coordinates` with `this._getCoordinates()`